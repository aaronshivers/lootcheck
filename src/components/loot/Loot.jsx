import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchBitcoin } from '../../actions/bitcoin';

export const Loot = ({ fetchBitcoin: fetchBitcoinFromProps, bitcoin, balance }) => {
  useEffect(() => {
    fetchBitcoinFromProps();
  });

  const computeBitcoin = () => {
    if (Object.keys(bitcoin).length === 0) return '';

    return balance / Number(bitcoin.bpi.USD.rate.replace(/,/g, ''));
  };

  return (
    <div>
      <h2>
        {['Bitcoin Balance: ', computeBitcoin()]}
      </h2>
    </div>
  );
};

Loot.defaultProps = {
  fetchBitcoin: null,
};

Loot.propTypes = {
  fetchBitcoin: PropTypes.func,
  bitcoin: PropTypes.exact({
    bpi: PropTypes.object,
    chartName: PropTypes.string,
    disclaimer: PropTypes.string,
    time: PropTypes.object,
  }).isRequired,
};

const mapStateToProps = (state) => state;
const mapDispatchToProps = { fetchBitcoin };

export default connect(mapStateToProps, mapDispatchToProps)(Loot);
