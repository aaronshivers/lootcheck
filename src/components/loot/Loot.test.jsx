import React from 'react';
import {
  mount,
  shallow,
} from 'enzyme';
import { Loot } from './Loot';

describe('Loot', () => {
  const mockFetchBitcoin = jest.fn();
  const balance = 10;
  let bitcoin = {};
  const fetchBitcoin = mockFetchBitcoin;

  let loot = shallow(
    <Loot
      balance={balance}
      bitcoin={bitcoin}
    />,
  );

  it('should render correctly', () => {
    expect(loot).toMatchSnapshot();
  });

  describe('when mounted', () => {
    beforeEach(() => {
      loot = mount(<Loot
        balance={balance}
        bitcoin={bitcoin}
        fetchBitcoin={fetchBitcoin}
      />);
    });

    it('should dispatch the `fetchBitcoin()` method it receives from props',
      () => {
        expect(mockFetchBitcoin).toHaveBeenCalled();
      });

    describe('when there are valid bitcoin props', () => {
      beforeEach(() => {
        bitcoin = { bpi: { USD: { rate: '1000' } } };
        loot = shallow(
          <Loot
            balance={balance}
            bitcoin={bitcoin}
          />,
        );
      });

      it('should display the correct bitcoin value', () => {
        expect(loot.find('h2').text())
          .toEqual('Bitcoin Balance: 0.01');
      });
    });
  });
});
