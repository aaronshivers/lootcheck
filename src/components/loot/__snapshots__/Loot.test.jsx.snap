// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot should render correctly 1`] = `
<div>
  <h2>
    Bitcoin Balance: 
  </h2>
</div>
`;
